### PR TITLE
Milestone 1 (Ruby): Bubble Sort

### PR DESCRIPTION
**Pull request by OdongAlican**
### This project was created using Technologies below
- Ruby
- Stickler
- Rubocop

### CODE REVIEW

**Review by whiz25 — CHANGES_REQUESTED** [review]
## Nice Progress :+1: 
You have done a wonderful job with the first attempt at this project. We, however, need to work on some failing edge cases before we can merge this PR.
- [ ] make your README more professional. Add more sections to your file. Use [this](https://github.com/microverseinc/readme-template) template for guidance
- [ ] bubble_sort method is implemented and working **(covered by automated tests)** ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_bubble_sort/sort.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-1)). This edge case is failing.
- [ ] bubble_sort_by method is implemented and working **(covered by automated tests)** ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_bubble_sort/sortBy.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-1)). This edge case is also failing.
- [ ] Fix the stickler errors
- [ ] check my comments on individual methods for more suggestions.

### Martin Afani

File: bubble_sort.rb [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+      if sorted == -1
+        arr.insert(i, x, y)
+      else
+        arr.insert(i, y, x)
+      end
+      i += 1
+    end
+  end
+  puts arr.to_s
+end
```
Comment (whiz25) [inline bubble_sort.rb]: - [ ] sorts an array by accepting a block
`Below is a sample test case that should work if `*bubble_sort_by*` is well implemented. Use it as a reference to make your method work as expected.`
```ruby
unsorted = (1..10).to_a.reverse!
bubble_sort_by(unsorted) { |left, right| left.length - right.length } == %w[hi hey hello] # should return true
```

File: bubble_sort.rb [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+      next unless value[j] < value[j + 1]
+
+      switch = value[j]
+      value[j] = value[j + 1]
+      value[j + 1] = switch
+    end
+  end
+
+  print value
+end
```
Comment (whiz25) [inline bubble_sort.rb]: - [ ] sorts an array of numbers in increasing order
```ruby
unsorted = (1..10).to_a.reverse!
bubble_sort(unsorted) == unsorted.sort # should return true
```

**Comment by ghost** [comment]
Fixed the requested changes.

**Review by kobiyoyo — CHANGES_REQUESTED** [review]
# Project Approved :thumbsup:
## Hello Team ,  You have done a great job with this project. :clap:

**Review by kobiyoyo — APPROVED** [review]
Approved
